### PR TITLE
fix: Consider all parents as scroll parents

### PR DESCRIPTION
**Pull request by ling1726**
This PR introduces `listScrollParents` which will collect every scroll parent in DOM of both target and container.

This behaviour already existed in poppver v2 https://github.com/floating-ui/floating-ui/blob/v2.x/src/dom-utils/listScrollParents.js

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Avatar-mount_base.html">627</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Avatar-mount.html">654</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Button-mount_base.html">316</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Button-mount.html">322</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Field-mount_base.html">1096</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Field-mount.html">1064</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProvider-mount_base.html">705</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProvider-mount.html">720</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProviderWithTheme-mount_base.html">77</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProviderWithTheme-mount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">66</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">68</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">71</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/InfoButton-mount_base.html">15</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/InfoButton-mount.html">11</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/MakeStyles-mount_base.html">866</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/MakeStyles-mount.html">889</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Persona-mount_base.html">1759</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/Persona-mount.html">1736</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/SpinButton-mount_base.html">1368</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29378/perf-test-react-components/SpinButton-mount.html">1335</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: apps/vr-tests-react-components/src/stories/Positioning.stories.tsx [hunk 1]
```diff
@@ -1084,6 +1084,47 @@ const ScrollJumpContext = () => {
   );
```
Comment (fabricteam) [inline apps/vr-tests-react-components/src/stories/Positioning.stories.tsx:1084]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=29378&commitId=1ce0a3dcdce67286e27d112b34bd8a0f14618839&env=prod&clientType=FLUENTUI)


  <details><summary><b>Positioning</b> 2 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|Positioning.Multiple scroll parents.chromium.png|0| Added|
|Positioning.Multiple scroll parents.container attached to target.chromium.png|0| Added|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-positioning/src/utils/listScrollParents.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { getScrollParent } from './getScrollParent';
+
+export function listScrollParents(node: HTMLElement): HTMLElement[] {
```
Comment (layershifter) [inline packages/react-components/react-positioning/src/utils/listScrollParents.ts:3]: Please a few unit tests for this function.

File: apps/vr-tests-react-components/src/stories/Positioning.stories.tsx [hunk 3]
```diff
@@ -1084,6 +1084,47 @@ const ScrollJumpContext = () => {
...
+            </button>
+          </div>
+        </div>
+      </div>
+      <div ref={containerRef} style={{ border: '2px solid blue', padding: 20, backgroundColor: 'white' }}>
+        Popover
+      </div>
+    </>
+  );
+};
```
Comment (layershifter) [inline apps/vr-tests-react-components/src/stories/Positioning.stories.tsx:1126]:
```suggestion
const MultiScrollParent = () => {
  const { targetRef, containerRef } = usePositioning({});
  const scrollContainerRef = React.useRef<HTMLDivElement>(null);

  const scroll = () => {
    if (scrollContainerRef.current) {
      scrollContainerRef.current.scrollBy({ top: 150 });
    }
  };

  return (
    <>
      <span>The popover should stay attached to the trigger</span>
      <button id="scroll" onClick={scroll}>
        scroll
      </button>
      <div
        ref={scrollContainerRef}
        style={{
          border: '2px dashed green',
          height: 300,
          width: 400,
          overflow: 'auto',
          display: 'grid',
          gridTemplateColumns: '350px auto',
          gridTemplateRows: '800px',
        }}
      >
        <div style={{ position: 'relative' }}>
          <div
            style={{
              overflow: 'auto',
              border: '2px dashed red',
              position: 'absolute',
              top: 150,
              left: 100,
              padding: 20,
            }}
          >
            <button id="target" ref={targetRef}>
              Trigger
            </button>
          </div>
        </div>
        <div
          style={{
            backgroundImage: 'linear-gradient(to bottom, #ffffff, #b9b9b9, #777777, #3b3b3b, #000000)',
          }}
        />
      </div>
      <div ref={containerRef} style={{ border: '2px solid blue', padding: 20, backgroundColor: 'white' }}>
        Popover
      </div>
    </>
  );
};
```

Let's make the story clearer and an indicator for the scroll (please check that it still catches the regression)

**Review by layershifter — APPROVED** [review]

**Pull request merged**